### PR TITLE
feat(issueMatch): Implement IssueMatch creation on ComponentVersion attachments

### LINKED ISSUE #314: feat(components): Implement ComponentVersion event emission and IssueMatch creation
**Task Description**
Implement event emission functionality for `ComponentVersion` attachments to issues and create corresponding IssueMatches for related `ComponentInstances`.

The system needs to:
1. Emit events when a `ComponentVersion` is attached to an issue
2. Process these events to create `IssueMatches` for all `ComponentInstances` that are associated with the attached `ComponentVersion`

**Acceptance Criteria:**

- [x] Implement event emission when a ComponentVersion is attached to an issue
- [x] Create an event handler to process ComponentVersion attachment events
- [x] For each ComponentInstance associated with the ComponentVersion, create a corresponding IssueMatch
- [x] Ensure proper error handling and logging throughout the process

### PR DESCRIPTION
**Pull request by dorneanu**
## Description

Each time a `ComponentVersion` is attached to an `Issue` (which at the moment happens when the `k8s-assets` scanner is running) we need to make sure that new `IssueMatches` get created. Therefore whenever a `AddComponentVersionToIssueEvent` is emitted we need to:

- fetch all `ComponentInstances` (based on `ComponentVersionId` from event)
- fetch all `IssueVariants` (basesd on `IssueId` from event)
- check if `IssueMatches` already exist (based on `ComponentInstanceId` and `IssueId`)
- if not create new `IssueMatches`

Also the `GetTargetRemediationTimeline` functionality has been moved into a `shared` package making it accesible from multiples packages (within `internal/app`). 

## What type of PR is this? (check all applicable)

- [ ] 🍕 Feature
- [ ] 🐛 Bug Fix
- [ ] 📝 Documentation Update
- [ ] 🎨 Style
- [x] 🧑‍💻 Code Refactor
- [ ] 🔥 Performance Improvements
- [x] ✅ Test
- [ ] 🤖 Build
- [ ] 🔁 CI
- [ ] 📦 Chore (Release)
- [ ] ⏩ Revert

## Related Tickets & Documents

- Closes #314

## Added tests?

- [x] 👍 yes
- [ ] 🙅 no, because they aren't needed
- [ ] 🙋 no, because I need help
- [ ] Separate ticket for tests # (issue/pr)

Tests related to the events emissions are in `issue_handler_events_test.go`. Maybe we should follow the same structure for other handlers as well (as the `_test` files are already quite big already; splitting by functionality might be a good idea). 

## Added to documentation?

- [ ] 📜 README.md
- [ ] 🤝 Documentation pages updated
- [x] 🙅 no documentation needed
- [ ] (if applicable) generated OpenAPI docs for CRD changes

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: internal/app/issue/issue_handler_events_test.go [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+			// Setup mock expectations for happy path
+			db.On("GetComponentInstances", &entity.ComponentInstanceFilter{
+				ComponentVersionId: []*int64{&componentVersion.Id},
+			}).Return([]entity.ComponentInstance{componentInstance}, nil)
+
+			db.On("GetIssueVariants", &entity.IssueVariantFilter{
+				IssueId: []*int64{&issueEntity.Id},
+			}).Return([]entity.IssueVariant{issueVariant}, nil)
+		})
+
```
Comment (drochow) [inline internal/app/issue/issue_handler_events_test.go:66]: Probably we want to add a test that the severity of the highest priority is taken

File: internal/app/shared/remediation.go [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+}
+
+// DefaultRemediationConfig provides default values for remediation timelines
+var DefaultRemediationConfig = RemediationTimelineConfig{
+	LowDays:      180, // 6 months
+	MediumDays:   90,  // 3 months
+	HighDays:     20,
+	CriticalDays: 7,
+	DefaultDays:  365, // 1 year
+}
```
Comment (drochow) [inline internal/app/shared/remediation.go:28]: I like it, maybe we can make in addition a comment to make them configurable via env variables in future

File: internal/app/issue/issue_handler_events.go [hunk 3]
```diff
@@ -89,3 +93,82 @@ type ListIssueNamesEvent struct {
...
+		}
+
+		if len(issueVariants) == 0 {
+			l.Warn("No issue variants found for issue")
+			return
+		}
+
+		// TODO: Use the first variant's severity (we could implement more complex severity selection logic here)
+		severity := issueVariants[0].Severity
+
```
[the remark this answers is not in the document]
Comment (MR2011) [reply internal/app/issue/issue_handler_events.go:133]: I'd prefer to implement this in this PR

[the remark this answers is not in the document]
Comment (drochow) [reply internal/app/issue/issue_handler_events.go:133]: I would also prefere this in this PR.

[the remark this answers is not in the document]
Comment (dorneanu) [reply internal/app/issue/issue_handler_events.go]: So far the update: 

- I've added IssueId to `ServiceIssueVariantFilter` (https://github.com/cloudoperators/heureka/blob/dorneanu/issue-314/Events-for-ComponentVersion/internal/entity/issue_variant.go#L65)
- I've changed the query to take IssueId into consideration (https://github.com/cloudoperators/heureka/blob/dorneanu/issue-314/Events-for-ComponentVersion/internal/database/mariadb/service_issue_variant.go#L38)
  - Not sure if this is sufficient
- `BuildIssueVariantMap` should be a shared functionality (https://github.com/cloudoperators/heureka/blob/dorneanu/issue-314/Events-for-ComponentVersion/internal/app/shared/issueVariant.go#L21)
  - in `issue_match_handler_events.go`(https://github.com/cloudoperators/heureka/blob/dorneanu/issue-314/Events-for-ComponentVersion/internal/app/issue_match/issue_match_handler_events.go#L154)
  - in `issue_handler_events.go` (https://github.com/cloudoperators/heureka/blob/dorneanu/issue-314/Events-for-ComponentVersion/internal/app/issue/issue_handler_events.go#L121) 

`CreateIssueMatches` (based on a `ComponentInstance`Id and some IssueVariants) should also be a shared functionality (it pretty much does the same, doesn't it?):

- For `issue_handler_events.go`: https://github.com/cloudoperators/heureka/blob/dorneanu/issue-314/Events-for-ComponentVersion/internal/app/issue/issue_handler_events.go#L143
- For `issue_match_handler_events.go`: https://github.com/cloudoperators/heureka/blob/dorneanu/issue-314/Events-for-ComponentVersion/internal/app/issue_match/issue_match_handler_events.go#L164-L206

The tests in `issue_match_handler_test.go` still need to be adapated. 

Please review so far and tell me if I'm on the right path.

[the remark this answers is not in the document]
Comment (dorneanu) [reply internal/app/issue/issue_handler_events.go]: I'll adapt now the tests.

File: internal/app/issue/issue_handler_events.go [hunk 4]
```diff
@@ -89,3 +93,82 @@ type ListIssueNamesEvent struct {
...
+		for _, compInst := range componentInstances {
+			l.WithField("event-step", "GetIssueMatches").Debug("Fetching issue matches related to Component Instance")
+			issue_matches, err := db.GetIssueMatches(&entity.IssueMatchFilter{
+				ComponentInstanceId: []*int64{&compInst.Id},
+				IssueId:             []*int64{&attachmentEvent.IssueID},
+			})
+
+			if err != nil {
+				l.WithField("event-step", "GetIssueMatches").WithError(err).Error("Error while fetching issue matches related to Component Instance")
+				return
```
[the remark this answers is not in the document]
Comment (dorneanu) [reply internal/app/issue/issue_handler_events.go]: Yep, you're right. Replaced by `continue`.

File: internal/app/issue/issue_handler_events.go [hunk 5]
```diff
@@ -89,3 +93,82 @@ type ListIssueNamesEvent struct {
...
+				ComponentInstanceId:   compInst.Id,
+				IssueId:               attachmentEvent.IssueID,
+				TargetRemediationDate: shared.GetTargetRemediationTimeline(severity, time.Now(), nil),
+			}
+			l.WithField("event-step", "CreateIssueMatch").WithField("issueMatch", issue_match).Debug("Creating Issue Match")
+
+			_, err = db.CreateIssueMatch(issue_match)
+			if err != nil {
+				l.WithField("event-step", "CreateIssueMatch").WithError(err).Error("Error while creating issue match")
+				return
```
[the remark this answers is not in the document]
Comment (dorneanu) [reply internal/app/issue/issue_handler_events.go]: Yep, you're right. Replaced by `continue`.

File: internal/app/issue/issue_handler_events.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (dorneanu) [reply internal/app/issue/issue_handler_events.go]: Locally tests are passing. Hopefully also in the CI/CD.

File: internal/app/issue/issue_handler_events_test.go (hunk 1 above)
Comment (dorneanu) [reply internal/app/issue/issue_handler_events_test.go:85]: Hmmm... Do you then agree that all tests [here](https://github.com/cloudoperators/heureka/blob/dorneanu/issue-314/Events-for-ComponentVersion/internal/app/issue_match/issue_match_handler_test.go#L361-L496) have to be moved to the [shared](https://github.com/cloudoperators/heureka/blob/dorneanu/issue-314/Events-for-ComponentVersion/internal/app/shared/issueVariant.go#L4) package?

File: internal/app/issue/issue_handler_events.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (dorneanu) [reply internal/app/issue/issue_handler_events.go]: All tests ran here also successfully.

[the remark this answers is not in the document]
Comment (dorneanu) [reply internal/app/issue/issue_handler_events.go]: Fixed unit tests. Keep in mind: Whenever you setup a `ServiceIssueVariantFilter` you always need specify the IssueId and ComponentId:

```go
filter := &entity.ServiceIssueVariantFilter{
  ComponentInstanceId: []*int64{lo.ToPtr(componentInstances[0].Id.Int64)},
  IssueId: []*int64{lo.ToPtr(issue.Id.Int64)},
}
					
```

**Review by drochow — APPROVED** [review]
LGTM! Good Job

**Review by MR2011 — APPROVED** [review]

**Pull request merged**